crates/red_knot_python_semantic/src/types.rs

```diff
@@ -890,23 +890,18 @@ impl<'db> Type<'db> {
                 .iter()
                 .any(|&elem_ty| self.is_subtype_of(db, elem_ty)),
 
-            (_, Type::TypeVar(typevar)) => match typevar.bound_or_constraints(db) {
-                // No types are a subtype of a bounded typevar, or of an unbounded unconstrained
-                // typevar, since there's no guarantee what type the typevar will be specialized
-                // to. If the typevar is bounded, it might be specialized to a smaller type than
-                // the bound. (This is true even if the bound is a final class, since the typevar
-                // can still be specialized to `Never`.)
-                None => false,
-                Some(TypeVarBoundOrConstraints::UpperBound(_)) => false,
-                // If the typevar is constrained, there must be multiple constraints, and the
-                // typevar might be specialized to any one of them. However, the constraints do not
-                // have to be disjoint, which means an lhs type might be a subtype of all of the
-                // constraints.
-                Some(TypeVarBoundOrConstraints::Constraints(constraints)) => constraints
-                    .elements(db)
-                    .iter()
-                    .all(|constraint| self.is_subtype_of(db, *constraint)),
-            },
+            // If the typevar is constrained, there must be multiple constraints, and the typevar
+            // might be specialized to any one of them. However, the constraints do not have to be
+            // disjoint, which means an lhs type might be a subtype of all of the constraints.
```

Comment:
This comment is true, but I have a hard time understanding how it's relevant to the code below. Would we write this arm any differently if constraints had to be disjoint? It would still be the case that the lhs type must be a subtype of _all_ constraints in order to be a subtype of the constrained typevar, which seems like the more important point to make here.

Reply:
My thinking was that the possibility of non-disjoint constraints is the only reason that this arm is needed. If the constraints are disjoint, I don't think there's any lhs that could be a subtype of all of them, is there?

Reply:
Oh! Yes, that totally makes sense :) Disregard this comment.